src/main/kotlin/bsu/cc/parser/XlsxParser.kt

```diff
@@ -0,0 +1,152 @@
+package bsu.cc.parser
+
+import org.apache.poi.ss.usermodel.Cell
+import org.apache.poi.ss.usermodel.CellType
+import org.apache.poi.ss.usermodel.Row
+import org.apache.poi.ss.usermodel.Sheet
+import org.apache.poi.xssf.usermodel.XSSFWorkbook
+import java.io.File
+import java.io.FileInputStream
+import java.lang.IllegalArgumentException
+import java.lang.IllegalStateException
+import java.lang.reflect.Type
+import java.util.*
+
+
+
+fun readWorkbook(fileName: String): XSSFWorkbook {
+    val file = FileInputStream(File(fileName))
+    return XSSFWorkbook(file)
+}
+
+fun <T> sheetToDataClasses(
+        sheet: Sheet,
+        dataProducer: (namedRowMap: Map<String, Cell>) -> T,
+        ignoreDuplicateHeaders: Boolean = false
+): Sequence<T> {
+    val headerMap = sheetHeaderMap(sheet)
+    fun translatedProducer(indexedRowMap: Map<Int, Cell>): T {
+        return dataProducer(indexedToNamedRowMap(indexedRowMap, headerMap, ignoreDuplicateHeaders))
+    }
+    return sheetToDataClasses(sheet, ::translatedProducer, true)
+}
+
+fun <T> sheetToDataClasses(
+        sheet: Sheet,
+        dataProducer: (indexedRowMap: Map<Int, Cell>) -> T,
+        excludeHeader: Boolean? = null
```

Comment:
what is the benefit to using a nullable value for excludeHeader here?

Reply:
It initially was used as ternary value, now it's a bit of a hack to make the `sheetToDataClass` method's JVM signatures unique, since the typing on the lambda is insufficient.